src/lib/MapPreview.svelte

```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
+  import Fa from 'svelte-fa/src/fa.svelte'
+  import { faPlay } from '@fortawesome/free-solid-svg-icons'
+
+  export let mapId: string
+
+  export let setPreviewKey: ((key: string | null) => void) | undefined = undefined
+</script>
+
+<a title="Preview Map" href="https://allpoland.github.io/ArcViewer/?id={mapId}" class="preview-map"
+  on:click={(e) => {
+    e.preventDefault()
+
+    // @ts-ignore
+    setPreviewKey?.call(this, mapId)
```

Comment:
Not even sure if binding it to `this` brings any benefits. Seems very "react-y".